src/rgw/rgw_auth.cc

```diff
@@ -416,7 +510,10 @@ void rgw::auth::WebIdentityApplier::load_acct_info(const DoutPrefixProvider* dpp
 
   //Check if user_id.buckets already exists, may have been from the time, when shadow users didnt exist
   RGWStorageStats stats;
-  int ret = user->read_stats(dpp, null_yield, &stats);
+  ceph::real_time last_synced;
+  ceph::real_time last_updated;
+  int ret = driver->load_stats(dpp, null_yield, federated_user, stats,
+                               last_synced, last_updated);
```

Comment:
@cbodley The part below where `create_account` needs to be looked at - a shadow user is created in rgw automatically derived from the `sub` field of the JWT, in case of AssumeRoleWithWebIdentity (for the first time), and stats are maintained for that user going forward. If resources are going to be tied to an account, then we might have to do away with these shadow users, if AssumeRoleWithWebIdentity is called for a role in an account.

Reply:
for account roles, it's probably fine for WebIdentityApplier to stop writing shadow users to rados as long as load_acct_info() still initializes the RGWUserInfo for s->user which lots of ops still rely on

it also looks like there's more to do here to expose the role's account id to WebIdentityApplier so it can handle account ownership in get_aclowner() and is_owner_of()

for RemoteApplier, we don't define any mapping between remote users and rgw accounts, so any shadow users we create wouldn't belong to an account. i suppose an admin could come in and add the shadow users to an account if they wanted with `radosgw-admin user modify --account-id`. so if RemoteApplier::load_acct_info() finds a user with an account id, we should use that to handle account ownership in get_aclowner() and is_owner_of()

Reply:
> for account roles, it's probably fine for WebIdentityApplier to stop writing shadow users to rados as long as load_acct_info() still initializes the RGWUserInfo for s->user which lots of ops still rely on
> 

if we stop creating shadow users for AssumeRoleWithWebIdentity, and we still need s->user, then what will the s->user point to, in other words, which user will we be referring to if we do away with the shadow users.

> it also looks like there's more to do here to expose the role's account id to WebIdentityApplier so it can handle account ownership in get_aclowner() and is_owner_of()
> 
> for RemoteApplier, we don't define any mapping between remote users and rgw accounts, so any shadow users we create wouldn't belong to an account. i suppose an admin could come in and add the shadow users to an account if they wanted with `radosgw-admin user modify --account-id`. so if RemoteApplier::load_acct_info() finds a user with an account id, we should use that to handle account ownership in get_aclowner() and is_owner_of()

Reply:
WebIdentityApplier::create_account() is initializing some RGWUserInfo and writing it with store_user(). i'm guessing we could skip the store_user() part and expose that RGWUserInfo as-is

Reply:
ok, yes, this will work as long as we do not need the bucket and user quota stats.